ext/json/tests/json_encode_pretty_print2.phpt

```diff
@@ -0,0 +1,55 @@
+--TEST--
+json_encode() with JSON_PRETTY_PRINT on declared properties
+--FILE--
+<?php
+class MyClass {
+    public $x;
+    public $y;
+    public function __construct($x = 123, $y = []) {
+        $this->x = $x;
+        $this->y = $y;
+    }
+}
+
+class HasNoProperties {}
+
+echo json_encode(new HasNoProperties()), "\n";
+echo json_encode(new HasNoProperties(), JSON_PRETTY_PRINT), "\n";
+
+echo json_encode(new MyClass()), "\n";
+echo json_encode(new MyClass(), JSON_PRETTY_PRINT), "\n";
+$obj = new MyClass();
+$obj->dynamic = new MyClass(null, []);
+echo json_encode($obj), "\n";
+echo json_encode($obj, JSON_PRETTY_PRINT), "\n";
+$obj = new MyClass();
+unset($obj->y);
+echo json_encode($obj), "\n";
+echo json_encode($obj, JSON_PRETTY_PRINT), "\n";
+unset($obj->x);
+echo json_encode($obj), "\n";
+echo json_encode($obj, JSON_PRETTY_PRINT), "\n";
+?>
+--EXPECT--
+{}
```

Comment:
The output in php 8.1.0-dev after this patch matches what I'm seeing in php 7.4

I think the original patch was worth it to avoid an increase in memory usage caused by json_encode(), and shouldn't cause any invalid memory accesses - there may be a tiny change in output for edge cases involving JsonSerializable and regular objects, but I don't expect code to rely on that behavior